test/seqan3-test.cmake

```diff
@@ -48,22 +48,40 @@ include (FindPackageMessage)
 set (SEQAN3_BENCHMARK_CLONE_DIR "${PROJECT_BINARY_DIR}/vendor/benchmark")
 set (SEQAN3_TEST_CLONE_DIR "${PROJECT_BINARY_DIR}/vendor/googletest")
 
-# required flags, includes, definitions and libraries for seqan3
-set (SEQAN3_STRICT_CXX_FLAGS "-pedantic" "-Werror" "-Wall" "-Wextra")
-
-# required flags, includes and libraries for seqan3/test/performance
-set (SEQAN3_BENCHMARK_CXX_FLAGS ${SEQAN3_STRICT_CXX_FLAGS})
-set (SEQAN3_BENCHMARK_INCLUDE_DIRS "")
-set (SEQAN3_BENCHMARK_LIBRARIES "")
-list (APPEND SEQAN3_BENCHMARK_INCLUDE_DIRS "${SEQAN3_BENCHMARK_CLONE_DIR}/include/")
-list (APPEND SEQAN3_BENCHMARK_INCLUDE_DIRS "${SEQAN3_CLONE_DIR}/test/include/")
-
-# required flags, includes and libraries for seqan3/test/unit
-set (SEQAN3_TEST_CXX_FLAGS ${SEQAN3_STRICT_CXX_FLAGS})
-set (SEQAN3_TEST_INCLUDE_DIRS "")
-set (SEQAN3_TEST_LIBRARIES "")
-list (APPEND SEQAN3_TEST_INCLUDE_DIRS "${SEQAN3_TEST_CLONE_DIR}/googletest/include/")
-list (APPEND SEQAN3_TEST_INCLUDE_DIRS "${SEQAN3_CLONE_DIR}/test/include/")
+# needed for add_library (seqan3::test::* INTERFACE IMPORTED)
+# see cmake bug https://gitlab.kitware.com/cmake/cmake/issues/15052
+file(MAKE_DIRECTORY ${SEQAN3_BENCHMARK_CLONE_DIR}/include/)
+file(MAKE_DIRECTORY ${SEQAN3_TEST_CLONE_DIR}/googletest/include/)
+
+# seqan3::test exposes a base set of required flags, includes, definitions and
+# libraries qhich are in common for **all** seqan3 tests
+add_library (seqan3::test INTERFACE IMPORTED)
+set_property (TARGET seqan3::test APPEND PROPERTY INTERFACE_COMPILE_OPTIONS "-pedantic"  "-Wall" "-Wextra" "-Werror")
+set_property (TARGET seqan3::test APPEND PROPERTY INTERFACE_LINK_LIBRARIES "seqan3::seqan3" "pthread")
+set_property (TARGET seqan3::test APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES "${SEQAN3_CLONE_DIR}/test/include/")
```

Comment:
Mhmm I am a little bit sceptical of putting this into the global seqan3-test configuration. 
What would be the benefit of knowing these targets in all build projects? 
There is no reuse within the target specific builds, is there?